Mathlib/NumberTheory/RamificationInertia/Basic.lean

```diff
@@ -894,12 +895,37 @@ theorem sum_ramification_inertia (K L : Type*) [Field K] [Field L] [IsDedekindDo
       algebraMap_injective_of_field_isFractionRing R S K L, le_bot_iff]
   · exact finrank_quotient_map p K L
 
+theorem inertiaDeg_le_finrank [NoZeroSMulDivisors R S] {p : Ideal R} [p.IsMaximal] (hp0 : p ≠ ⊥)
+    (P : Ideal S) [hP₁ : P.IsPrime] [hP₂ : P.LiesOver p] :
+    p.inertiaDeg P ≤ Module.finrank K L := by
+  classical
+  have hP : P ∈ primesOverFinset p S := (mem_primesOverFinset_iff hp0 _).mpr ⟨hP₁, hP₂⟩
+  rw [← sum_ramification_inertia S K L hp0, ← Finset.add_sum_erase _ _ hP]
+  refine le_trans (Nat.le_mul_of_pos_left _ ?_) (Nat.le_add_right _ _)
+  exact Nat.pos_iff_ne_zero.mpr <| IsDedekindDomain.ramificationIdx_ne_zero_of_liesOver _ hp0
+
+theorem ramificationIdx_le_finrank [NoZeroSMulDivisors R S] {p : Ideal R} [p.IsMaximal]
+    (hp0 : p ≠ ⊥) (P : Ideal S) [hP₁ : P.IsPrime] [hP₂ : P.LiesOver p] :
+    p.ramificationIdx (algebraMap R S) P ≤ Module.finrank K L := by
+  classical
+  have hP : P ∈ primesOverFinset p S := (mem_primesOverFinset_iff hp0 _).mpr ⟨hP₁, hP₂⟩
+  rw [← sum_ramification_inertia S K L hp0, ← Finset.add_sum_erase _ _ hP]
+  refine le_trans (Nat.le_mul_of_pos_right _ ?_) (Nat.le_add_right _ _)
+  exact Nat.pos_iff_ne_zero.mpr <|  inertiaDeg_ne_zero p P
```

Comment:
This is also true for `p = ⊥` by simp, hopefully.

Reply:
Yes. Thanks!